### DISCUSSION TITLE
Debugging in C for a newbie

### DISCUSSION THREAD

**Discussion by PremiereStoss** [LazyVim/LazyVim#2500] [General]
Hi all,

I could use some help.  I am learning C, Linux, and neovim all at the same time.  "GO BIG OR GO HOME" :)

I have been programming in C in neovim and I am really enjoying it.  With lazyvim, I have set up everything and it is all working except for debugging.  I don't know what I am missing, whether that is a plugin or if I just don't know how to work it.

Here is what I have.
Mason: DAP: codelldb, cpptools
LazyVim: nvim-dap, nvim-dap-ui, nvim-dap-virtual-text, treesitter, and many more.  Just put the ones I think are relevant.
Linux: gcc, gdb both are installed.

When I hit <leader>dc I get a configuration window
![image](https://github.com/LazyVim/LazyVim/assets/53406832/319175a1-bc2e-4132-bf60-65c2699fa740)
I hit launch file and get this...
![image](https://github.com/LazyVim/LazyVim/assets/53406832/2bd0689e-639b-4ed2-876e-73901f051b9c)

I believe this is what is causing my problems but I don't know what to do here.

Furthermore, if I got into dap-ui I just see the following.  I can't get anything to run in the debug window.

With <leader>d I can access debug menu.  I can set a breakpoint but after that, I am confused on what to do next.  I go into the debug ui and this is what I see.

![image](https://github.com/LazyVim/LazyVim/assets/53406832/5beb1e96-34e3-450e-9c2e-e95674018e70)

I am missing something but I don't know what.  Any help in pointing me in the right direction would be greatly appreciated.

Thanks!

**Comment by dpetka2001** [comment]
You should compile your program with debug symbols. Using `gcc` as your compiler, you could do this with `gcc -g -Wall main.c -o main` (if you use another compiler read the docs about compiling with debug symbols) and then on the popup menu you get about `Path to executable` just append `main` or whatever name you gave your executable. Then setting a breakpoint and doing `Launch file` should give you what you want.

**Reply by PremiereStoss** [reply]
---palm to the forehead---
@dpetka2001 you nailed it.  I can't believe I didn't think of that.  I have never used it but I have see it before.  Thank you so much!  I am a real newb to this whole and complete setup.  Baby steps, I guess.

Thank you, thank you!
